src/link/link.cpp

```diff
@@ -19,29 +19,28 @@ Link::Link(LinkType linkType, QString name, QObject* parent)
 {
     switch(linkType) {
     case LinkType::None :
-        _abstractLink.reset(new AbstractLink());
+        _abstractLink = new AbstractLink(this);
```

Comment:
The class was created without parent. I don't see the point here. This is also more error prone, E.g: Why are you not deleting `_abstractLink`?

Reply:
Also, I'm not aware of such thing as as crashing with Qt and smartpointers, should we stay with smartPointers and remove the setParent ? This appears to be the less error prone option IMO.